### ISSUE TITLE
Scala 3 compiler does not create same bytecode for java.util.Map[String, String] as Scala 2

### ISSUE THREAD

**Issue by ilx** [lampepfl/dotty#18900]
## Compiler version

3.2.0 and newer 

## Minimized example

```Scala
import java.util.{Date, List => JList, Map => JMap, Set => JSet}
class MapStringString { var prop: JMap[String, String] = _ }
val x = classOf[MapStringString]
val res0 = x.getDeclaredFields()(0)
res0.getGenericType
```

## Output

Scala 2.13 outputs:

```scala
val x: Class[MapStringString] = class MapStringString
val res0: java.lang.reflect.Field = private java.util.Map MapStringString.prop
val res9: java.lang.reflect.Type = java.util.Map<java.lang.String, java.lang.String>
```

while Scala 3 gives:

```
scala> val x = classOf[MapStringString]
val x: Class[MapStringString] = class rs$line$2$MapStringString

scala> x.getDeclaredFields()(0)
val res0: java.lang.reflect.Field = private java.util.Map rs$line$2$MapStringString.prop

scala> res0.getGenericType
val res1: java.lang.reflect.Type = interface java.util.Map
```

## Expectation

I would expect that Scala 3 returns same GenericType. 
Can you point me at the documentation where this and similar changes are documented?

**Comment by ilx** [comment]
It looks the same as https://github.com/lampepfl/dotty/issues/17069

**Comment by SethTisue** [comment]
still broken in 3.3.2, but fixed in 3.4.0
